src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
+            jsondata=None,
+            params=None,
+            response=response,
+        )
         check_for_api_error(response)
-        json_response = response.json()
+        json_response = cast(dict[str, Any], response.json())
         return json_response
 
-    def put(self, path: str, jsondata: Optional[str] = None, content_type: str = "application/json"):
-        """
-        Send data to a RESTful server using a HTTP PUT request
-        :param path: Path of RESTful route
-        :param jsondata: Valid JSON as string
-        :param content_type: HTTP Content-Type [default: 'application/json']
-        :return:
+    def put(
+        self,
+        path: str,
+        jsondata: Optional[str] = None,
+        content_type: str = "application/json",
+    ) -> dict[str, Any]:
         """
+        Make a HTTP GET request to the server to which this connection has been established.
 
-        if path[0] != "/":
+        Args:
+            path: Path of RESTful route
+            jsondata: Valid JSON as string
+            content_type: HTTP Content-Type [default: 'application/json']
+        """
+        # timeout must be None,
+        # otherwise the client can get a timeout error while the API is still processing the request
+        # in that case, the client's retry will fail, and the response of the original API call is lost
+        timeout = None
+        if not path.startswith("/"):
             path = "/" + path
-        if jsondata is None:
-            response = requests.put(
-                self._server + path,
-                headers={"Authorization": "Bearer " + self._token},
-                timeout=20,
-            )
-        else:
-            response = requests.put(
-                self._server + path,
-                headers={"Content-Type": content_type + "; charset=UTF-8", "Authorization": "Bearer " + self._token},
-                data=jsondata,
-                timeout=20,
-            )
+        headers = {}
+        if self.token:
+            headers["Authorization"] = "Bearer " + self.token
+        if jsondata:
+            headers["Content-Type"] = content_type + "; charset=UTF-8"
+
+        response = requests.put(
+            self.server + path,
+            headers=headers,
+            data=jsondata,
+            timeout=timeout,
+        )
+        self.write_log_file(
+            method="PUT",
+            route=path,
+            headers=headers,
+            jsondata=jsondata,
+            params=None,
+            response=response,
+        )
         check_for_api_error(response)
-        result = response.json()
-        return result
+        json_response = cast(dict[str, Any], response.json())
+        return json_response
 
-    def delete(self, path: str, params: Optional[any] = None):
-        """
-        Send a delete request using the HTTP DELETE request
-        :param path: Path of RESTful route
-        :return: Response from server
+    def delete(
```

Comment:
does DSP-Tools ever do a delete request? if not, this could be removed

Reply:
yes, there are delete requests

Reply:
let me re-phrase this:  
is any of the delete methods ever used? I had a quick look and didn't find any instances.  
I know it could be used by someone using DSP-Tools as a module for scripts... but do people use delete in these cases?

Reply:
To be sure that I understand you correctly: Connection.delete() is used by Group.delete(), which is only used in the e2e tests. 

If all usages of Connection.delete() are only used in the e2e tests, I could remove all usages, and modify the e2e tests. Is that what you mean?